xcube/cli/_gen2/callback.py

```diff
@@ -30,17 +35,48 @@
 
 
 class CallbackApiProgressObserver(ProgressObserver):
-    def __init__(self, callback_api_url: str):
-        self.callback_api_url = callback_api_url
+    def __init__(self, callback_api_cfg: Request):
+        assert_given(callback_api_cfg)
+        assert_given(callback_api_cfg.callback)
+        assert_condition(callback_api_cfg.callback.api_uri and callback_api_cfg.callback.access_token,
+                         "Both, api_uri and access_token must be given.")
+        self.callback_api_cfg = callback_api_cfg
+
+    def _send_request(self, event: str, state_stack: Sequence[ProgressState]):
+        assert_given(state_stack, "ProgressStates")
+        state_stack = state_stack[0].to_dict()
+        callback = {"event": event, "state": state_stack}
+
+        callback_api_uri = self.callback_api_cfg.callback.api_uri
+        callback_api_access_token = self.callback_api_cfg.callback.access_token
+
+        header = {"Authorization": f"Bearer {callback_api_access_token}"}
+        requests.put(callback_api_uri, json=callback, headers=header)
+
+    def on_begin(self, state_stack: Sequence[ProgressState]):
+        self._send_request("on_begin", state_stack)
+
+    def on_update(self, state_stack: Sequence[ProgressState]):
+        self._send_request("on_update", state_stack)
+
+    def on_end(self, state_stack: Sequence[ProgressState]):
+        self._send_request("on_end", state_stack)
+
+
+class CallbackTerminalProgressObserver(ProgressObserver):
```

Comment:
Would be nice to use a Python package for this purpose, e.g. https://tqdm.github.io/